src/helm/proxy/clients/anthropic_client.py

```diff
@@ -26,6 +26,18 @@ class AnthropicClient(Client):
     """
     Client for the Anthropic models (https://arxiv.org/abs/2204.05862).
     They use their own tokenizer.
+    Here are a list of bugs that we deal with in this client:
+    - The completions is often too verbse, so we add the PROMPT_ANSWER_START to the prompt.
+    - The completions often start with a colon, space, or newline, so we remove it. This means
+    that we need to query more tokens than necessary to ensure that the completion does not start
+    with a colon, space, or newline. We query `max_tokens + ADDITIONAL_TOKENS` tokens and then
+    remove the excess tokens at the end of the completion.
+    - The tokenizer returns stange characters like "Ġ". See TEST_TOKENS in test_anthropic_with_service.py.
+    It shows the actual tokens that are returned by the tokenizer.
+    - The API sometimes returns "Prompt must contain anthropic.AI_PROMPT" or Prompt length + max_tokens
+    exceeds max (9192). This is a bug related to the window_service that does not properly truncate some
+    prompts. The first issue is caused by the suffix being truncated and the second is something we do not
+    handle yet (we have not limit on prompt length + max_tokens).
```

Comment:
Seems like this are two separate issues? Could you split into two points?

Issue 1: AI prompt gets truncated
Issue 2: Window service does not respect max length

Could you also file an GitHub issue for issue 2? Have you looked into why this only happens for Anthropic and not other models?

Reply:
I think this is something we never tackled with?